---
title: "HW3 - Differential expression analysis"
output:
  html_document:
    df_print: paged
  pdf_document: default
---

```{r init, include=F}
library(knitr)
opts_chunk$set(message=FALSE, warning=FALSE, eval=TRUE, echo=TRUE, cache=TRUE)
.ex <- 1
library(ggplot2)
theme_set(theme_bw(base_size=16) + theme(strip.background = element_blank()))
```
Submitted by:  
ID1: 320983216
Name1: Maxim Kolchinsky  
ID2: 321340580
Name2: Anna Romanov

# The Biology

The data for this lesson comes from:  
> Saigi _et al_. "MET-Oncogenic and JAK2-Inactivating Alterations Are Independent Factors That Affect Regulation of PD-L1 Expression in Lung Cancer" _PLoS ONE_. [2018 Jun 13;9(6):e99625](https://webcourse.cs.technion.ac.il/236523/Winter2018-2019/hw/WCFiles/Q1_paper.pdf?5265). PMID: [24926665](https://www.ncbi.nlm.nih.gov/pubmed/29898990).

Purpose: The blockade of immune checkpoints such as PDL1 and PD-1 is being exploited therapeutically in several types
of malignancies. Here, we aimed to understand the contribution of the genetics of lung cancer to the ability of tumor cells to
escape immunosurveillance checkpoints.
Experimental Design: More than 150 primary non-small
cell lung cancers, including pulmonary sarcomatoid carcinomas, were tested for levels of the HLA-I complex, PD-L1,
tumor-infiltrating CD8þ lymphocytes, and alterations in main
lung cancer genes. Correlations were validated in cancer cell
lines using appropriate treatments to activate or inhibit selected pathways. We also performed RNA sequencing to assess
changes in gene expression after these treatments.
Results: MET-oncogenic activation tended to associate
with positive PD-L1 immunostaining, whereas STK11 mutations were correlated with negative immunostaining. In
MET-altered cancer cells, MET triggered a transcriptional
increase of PD-L1 that was independent of the IFNgmediated JAK/STAT pathway. The activation of MET also
upregulated other immunosuppressive genes (PDCD1LG2
and SOCS1) and transcripts involved in angiogenesis
(VEGFA and NRP1) and in cell proliferation. We also report
recurrent inactivating mutations in JAK2 that co-occur with
alterations in MET and STK11, which prevented the induction of immunoresponse-related genes following treatment
with IFNg.
Conclusions: We show that MET activation promotes the
expression of several negative checkpoint regulators of the
immunoresponse, including PD-L1. In addition, we report
inactivation of JAK2 in lung cancer cells that prevented the
response to IFNg. These alterations are likely to facilitate
tumor growth by enabling immune tolerance and may affect
the response to immune checkpoint inhibitors

# Data
This data was downloaded from GEO ([GSE:GSE109720](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE109720))  

### Import count data and metadata

```{r libs_importdata}
library(readr)
library(dplyr)
library(ggplot2)

rawcounts <- read_csv("data/lung_counts.csv")
metadata <-  read_csv("data/lung_metadata.csv")
```

### General information on the data

There are `r length(unique(metadata$celltype))` cell types in the dataset

The distribution of samples accross cell types is:
```{r celltype_info}
table(metadata$celltype)
```

The number of expressed genes per sample:
```{r expressed genes}
vals<- data.frame('expressed_genes' = colSums(rawcounts[,-1]>0))
vals
ggplot(vals, aes(x=expressed_genes)) + geom_histogram(bins=15)
```

```{r highly_expressed genes}
vals<- data.frame('highly_expressed_genes' = colSums(rawcounts[,-1]>1000))
vals
ggplot(vals, aes(x=highly_expressed_genes)) + geom_histogram(bins=15)
```

### Filter the data
```{r filter}
treatment1<- 'No treatment'
treatment2<- 'Crizotinib'
celltype0<- 'EBC1'
dex <- metadata$dex
filtered_metadata<- metadata %>% filter(dex==treatment1 | dex==treatment2 ) %>% filter(celltype==celltype0)
filtered_metadata

filtered_samples<- filtered_metadata$id
filtered_counts<- rawcounts %>% select(filtered_samples)
filtered_counts<- bind_cols(rawcounts[,'ensgene'],filtered_counts)
filtered_counts

```

# DESeq2 analysis

```{r loadDESeq2}
library(DESeq2)
```

### Importing data

first we fix the row names:
```{r fixing rownames}
filtered_countsRN <- data.frame(filtered_counts, row.names=1)
filtered_countsRN

filtered_metadataRN <- data.frame(filtered_metadata, row.names=1)
filtered_metadataRN

```

now we can construct the DESeq2 dataset
```{r constructDDS}
dds <- DESeqDataSetFromMatrix(countData=filtered_countsRN, colData=filtered_metadataRN, design=~dex)
dds
```


### Run the DESeq pipeline
We run the DESe12 pipline:
```{r deseq_pipeline}
dds <- DESeq(dds)
```


### Getting results
first look at the resuts
```{r getResults}
res <- results(dds, tidy=TRUE)
res <- tbl_df(res)
res
```

Using a `%>%`, `arrange` the results by the adjusted p-value.

```{r }
res <- res %>%
  arrange(padj)
res

```

```{r}
res_1 <- res %>% filter(padj<0.05) %>% nrow()
```

There `r res_1` are genes with p-value < 0.05

```{r}
res_2 <- res %>% filter(padj<0.01) %>% nrow()
```

There `r res_2` are genes with p-value < 0.01

```{r save results}
res %>%
  filter(padj<0.95) %>%
  write_csv("sigresults.csv")
```

# Data Visualization

### Plotting counts

```{r plotCounts1}
gene <- res$row[runif(1,0,10000)]
gene_name <- unlist(strsplit(gene,',', fixed = TRUE))[2]
plotCounts(dds, gene=gene, intgroup="dex",returnData=TRUE) %>% ggplot(aes(dex, count)) + geom_boxplot(aes(fill=dex)) + scale_y_log10() + ggtitle(gene_name)
```

### MA & Volcano plots

```{r}
# Create the new column
res <- res %>% mutate(sig=padj<0.95)

# How many of each?
res %>%
  group_by(sig) %>%
  summarize(n=n())

log2fold_above_2 = res %>% filter(log2FoldChange > 2) %>% nrow()
log2fold_below_minus_2 = res %>% filter(log2FoldChange < -2) %>% nrow()

res %>% ggplot(aes(baseMean, log2FoldChange, col=sig)) + geom_point() + scale_x_log10() + ggtitle("MA plot")
```

There are `r log2fold_above_2` with log fold change above 2

There are `r log2fold_below_minus_2` with log fold change below -2

```{r volcanoplot}
res %>% ggplot(aes(log2FoldChange, -1*log10(pvalue), col=sig)) + geom_point() + ggtitle("Volcano plot")
```

# Record `sessionInfo()`

The `sessionInfo()` prints version information about R and any attached packages. It's a good practice to always run this command at the end of your R session and record it for the sake of reproducibility in the future.

```{r sessionInfo}
sessionInfo()
```

